vmm/src/device_manager/legacy.rs

```diff
@@ -62,22 +68,36 @@ impl LegacyDeviceManager {
             com_evt_1_3.try_clone().map_err(Error::EventFd)?,
             Box::new(stdout()),
         )));
+        let second_serial = match second_serial {
```

Comment:
Instead of a match statement with a noop `None` branch, just use `map`

```rust
let second_serial = second_serial.map(|serial| {
    Arc::new(Mutex::new(devices::legacy::Serial::new_out( ... )))
});
```